tests/c_languages/CPPCheckBearTest.py

```diff
@@ -1,37 +1,50 @@
+import os
+import unittest
+
+from queue import Queue
+
 from bears.c_languages.CPPCheckBear import CPPCheckBear
-from coalib.testing.LocalBearTestHelper import verify_local_bear
-
-good_file = """
-using namespace std;
-int main() {
-    cout << "Hello, world!" << endl;
-    return 0;
-}"""
-
-warn_file = """
-void f1(struct fred_t *p)
-{
-    int x;
-    if (p)
-        do_something(x);
-}"""
-
-bad_file = """
-#define f(c) { \
-    char s[10]; \
-    s[c] = 42; \
-}
-int main() {
-    f(100);
-    return 0;
-}"""
-
-
-CPPCheckBearTest1 = verify_local_bear(CPPCheckBear,
-                                      valid_files=(good_file, warn_file),
-                                      invalid_files=(bad_file,))
-
-CPPCheckBearTest2 = verify_local_bear(CPPCheckBear,
-                                      valid_files=(good_file,),
-                                      invalid_files=(warn_file, bad_file),
-                                      settings={'enable': 'unusedFunction'})
+from coalib.testing.BearTestHelper import generate_skip_decorator
+from coalib.settings.Section import Section
+from coalib.settings.Setting import Setting
+
+
+def get_absolute_test_path(file):
+    return os.path.join(os.path.dirname(__file__),
+                        'cppcheck_test_files', file)
+
+
+@generate_skip_decorator(CPPCheckBear)
+class CPPCheckBearTest(unittest.TestCase):
+
+    def setUp(self):
+        self.section = Section('cppcheck')
+        self.section.append(Setting('enable', 'all'))
+        self.file_dict = {}
+        self.queue = Queue()
+        self.test_files = ['file1.cpp', 'file2.cpp', 'file3.c']
+
+    def get_results(self, files_to_check):
+        self.files = [get_absolute_test_path(file) for file in files_to_check]
+        for filename in self.files:
+            with open(filename, 'r', encoding='utf-8') as _file:
+                self.file_dict[filename] = tuple(_file.readlines())
+        self.uut = CPPCheckBear(self.file_dict,
+                                self.section,
+                                self.queue)
+        return list(self.uut.run_bear_from_section([], {}))
+
+    def test_results_complete(self):
+        results = self.get_results(self.test_files)
+        messages = [result.message for result in results]
+        self.assertTrue("The function 'fun' is never used" in messages[1])
```

Comment:
1. there's `assertIn`
2. Why not asserting for the complete message?
   ```
   self.assertEqual(message, '...')
   ```

Reply:
Can not use asserting for the complete message because there is little bit difference between `result.messages` given by TravisCI and CircleCI.In this case, there is a difference of trailing period in the messages.

Reply:
>1. there's `assertIn`

Do you want me to use `assertIn` in place of `assertTrue`